azurerm/helpers/azure/app_service.go

```diff
@@ -1114,16 +1142,55 @@ func FlattenAppServiceLogs(input *web.SiteLogsConfigProperties) []interface{} {
 				blobStorageItem["retention_in_days"] = *blobStorageInput.RetentionInDays
 			}
 
-			blobStorage = append(blobStorage, blobStorageItem)
+			// The API returns a non nil application logs object when other logs are specified so we'll check that this structure is empty before adding it to the statefile.
+			if blobStorageInput.SasURL != nil {
+				blobStorage = append(blobStorage, blobStorageItem)
+
+				appLogsItem["azure_blob_storage"] = blobStorage
+			}
 
-			appLogsItem["azure_blob_storage"] = blobStorage
 		}
 
-		appLogs = append(appLogs, appLogsItem)
+		if len(appLogsItem) > 0 {
+			appLogs = append(appLogs, appLogsItem)
+		}
 
 		result["application_logs"] = appLogs
```

Comment:
this now wants moving below the next `}` so that this is always set? (as [this](https://github.com/terraform-providers/terraform-provider-azurerm/pull/4025/files#diff-35e1b76513015f662d0cd675debc886fR1131)) wants moving up above the `if != nil` check